test/test_hyper.py

```diff
@@ -1282,6 +1342,40 @@ def data_callback(frame):
         assert isinstance(f, RstStreamFrame)
         assert f.error_code == 1 # PROTOCOL_ERROR
 
+    def test_connection_sends_rst_frame_if_frame_size_too_large(self):
+        sock = DummySocket()
+        d = DataFrame(1)
+        d.data = b'hi there frame'
+        sock.buffer = BytesIO(d.serialize())
+
+        def send_rst_frame(stream_id, error_code):
+            assert stream_id == 1
+            assert error_code == 6 #FRAME_SIZE_ERROR
+
+        c = HTTP20Connection('www.google.com')
+        # Lower the maximum frame size settings in order to force the
+        # connection to send a reset frame with FRAME_SIZE_ERROR
+        c._settings[SettingsFrame.SETTINGS_MAX_FRAME_SIZE] = 10
+        c._sock = sock
+        c._send_rst_frame = send_rst_frame
+        c.request('GET', '/')
+        c._recv_cb()
```

Comment:
This test might spuriously pass, because if `send_rst_frame` isn't called there will still be no failing asserts. Might be better to use a list that gets appended to, as I do elsewhere in this class.